File: python/sort_test.py
# -*- encoding:utf-8 -*-
"""
    author: nick
    date: 2019/6/1
"""

import random
import unittest

from python.selectSort import select_sort
from python.quickSort import quick_sort
from python.bubbleSort import bubble_sort

def get_random_arr():
    nums = [i for i in range(100)]
    test_arr = []
    for i in range(10):
        test_arr.append(random.choice(nums))
    return test_arr


class SortTest(unittest.TestCase):

    def sort_test(self, sorted_arr: list):
        for i in range(len(sorted_arr)-1):
            for j in range(i+1, len(sorted_arr)):
                self.assertLessEqual(sorted_arr[i], sorted_arr[j])

    def test_bubble_sort(self):
        test_arr = get_random_arr()
        sorted_arr = bubble_sort(test_arr)
        print(sorted_arr)
        self.sort_test(sorted_arr)

    def test_quick_sort(self):
        test_arr = get_random_arr()
        sorted_arr = quick_sort(test_arr)
        print(sorted_arr)
        self.sort_test(sorted_arr)

    def test_select_sort(self):
        test_arr = get_random_arr()
        sorted_arr = select_sort(test_arr)
        print(sorted_arr)
        self.sort_test(sorted_arr)


if __name__ == '__main__':
    unittest.main()

File: golang/sort_test.go
package golang

import (
	"fmt"
	"math/rand"
	"testing"
)

func GetRandomArray() (array []int) {
	array = make([]int, 0, 10)
	for i := 0; i < 10; i++ {
		randomNum := rand.Intn(100)
		array = append(array, randomNum)
	}
	return array
}

func SortTest(array []int) bool {
	for i := 0; i < len(array)-1; i++ {
		for j := i+1; j < len(array); j++ {
			if array[j] < array[i] {
				return false
			}
		}
	}
	return true
}

func TestBubbleSort(t *testing.T) {
	array := GetRandomArray()
	fmt.Println("random array", array)
	sortedArr := BubbleSort(array)
	fmt.Println("sorted array", sortedArr)
	if !SortTest(sortedArr) {
		t.Errorf("bubble sort err, before sort:%v, after sort:%v", array, sortedArr)
	}
}

func TestInsertSort(t *testing.T) {
	array := GetRandomArray()
	fmt.Println("random array", array)
	sortedArr := InsertSort(array)
	fmt.Println("sorted array", sortedArr)
	if !SortTest(sortedArr) {
		t.Errorf("bubble sort err, before sort:%v, after sort:%v", array, sortedArr)
	}
}

func TestMergeSort(t *testing.T) {
	array := GetRandomArray()
	fmt.Println("random array", array)
	sortedArr := MergeSort(array)
	fmt.Println("sorted array", sortedArr)
	if !SortTest(sortedArr) {
		t.Errorf("bubble sort err, before sort:%v, after sort:%v", array, sortedArr)
	}
}

func TestQuickSort(t *testing.T) {
	array := GetRandomArray()
	fmt.Println("random array", array)
	sortedArr := QuickSort(array)
	fmt.Println("sorted array", sortedArr)
	if !SortTest(sortedArr) {
		t.Errorf("bubble sort err, before sort:%v, after sort:%v", array, sortedArr)
	}
}

func TestSelectSort(t *testing.T) {
	array := GetRandomArray()
	fmt.Println("random array", array)
	sortedArr := SelectSort(array)
	fmt.Println("sorted array", sortedArr)
	if !SortTest(sortedArr) {
		t.Errorf("bubble sort err, before sort:%v, after sort:%v", array, sortedArr)
	}
}